sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -148,6 +149,8 @@ private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
 
 object SharedState {
 
+  URL.setURLStreamHandlerFactory(new SparkUrlStreamHandlerFactory())
```

Comment:
But with your wrap `SparkUrlStreamHandlerFactory`, why it can be called more than once? I didn't get the point here.